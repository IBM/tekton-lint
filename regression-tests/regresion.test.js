const rules = require('../rules');

it('regression tests', async () => {
  const result = await rules(['./regression-tests/*.yaml']);
  expect(result).toMatchSnapshot();
});
