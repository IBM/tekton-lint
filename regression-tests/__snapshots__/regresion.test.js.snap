// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regression tests 1`] = `
Array [
  Object {
    "level": "error",
    "message": "'my-duplicated-task' is already defined (as a 'Task'), it can't be redefined (as a 'Task')",
  },
  Object {
    "level": "error",
    "message": "'my-duplicated-pipeline' is already defined (as a 'Pipeline'), it can't be redefined (as a 'Pipeline')",
  },
  Object {
    "level": "error",
    "message": "'my-duplicated-triggerbinding' is already defined (as a 'TriggerBinding'), it can't be redefined (as a 'TriggerBinding')",
  },
  Object {
    "level": "error",
    "message": "'my-duplicated-eventlistener' is already defined (as a 'EventListener'), it can't be redefined (as a 'EventListener')",
  },
  Object {
    "level": "error",
    "message": "'my-duplicated-triggertemplate' is already defined (as a 'TriggerTemplate'), it can't be redefined (as a 'TriggerTemplate')",
  },
  Object {
    "level": "error",
    "message": "'pipeline-with-taskspec' is already defined (as a 'Pipeline'), it can't be redefined (as a 'Pipeline')",
  },
  Object {
    "level": "error",
    "message": "'pipeline-with-taskspec' is already defined (as a 'Pipeline'), it can't be redefined (as a 'Pipeline')",
  },
  Object {
    "level": "error",
    "message": "'pipeline-with-taskspec' is already defined (as a 'Pipeline'), it can't be redefined (as a 'Pipeline')",
  },
  Object {
    "level": "error",
    "message": "'task-without-volumes' is already defined (as a 'Task'), it can't be redefined (as a 'Task')",
  },
  Object {
    "level": "error",
    "message": "'my-pipeline-with-workspace' is already defined (as a 'Pipeline'), it can't be redefined (as a 'Pipeline')",
  },
  Object {
    "level": "error",
    "message": "'template-pipelineref-null' is already defined (as a 'TriggerTemplate'), it can't be redefined (as a 'TriggerTemplate')",
  },
  Object {
    "level": "error",
    "message": "Resource ServiceAccount 'tutorial-service' has an invalid 'resourceVersion' key in its resource definition.",
  },
  Object {
    "level": "warning",
    "message": "Task 'task-with-params-and-inputs' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Task 'task-without-params' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Task 'task-null-params' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Task 'task-without-volumes' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Task 'task-with-undefined-volume' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Task 'tekton-alpha-mixed' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Task 'my-alpha-task' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-params' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-without-params' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-null-params' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-name' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-missing-and-extra-params' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskspec' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-task-missing-runafter' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-task-with-null-runafter' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'tekton-alpha-mixed-pipeline' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'my-alpha-pipeline' is defined with apiVersion tekton.dev/v1alpha1, consider migrating to tekton.dev/v1beta1",
  },
  Object {
    "level": "error",
    "message": "Task 'tekton-alpha-mixed' is defined with apiVersion tekton.dev/v1alpha1, but defines spec.params. Use spec.inputs.params instead.",
  },
  Object {
    "level": "error",
    "message": "Task 'tekton-beta-mixed' is defined with apiVersion tekton.dev/v1beta1, but defined spec.inputs.params. Use spec.params instead.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'tekton-alpha-mixed-pipeline' is defined with apiVersion tekton.dev/v1alpha1, but defines an inlined task (my-task) with spec.params. Use spec.inputs.params instead.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'tekton-beta-mixed-pipeline' is defined with apiVersion tekton.dev/v1beta1, but defines an inlined task (my-task) with spec.inputs.params. Use spec.params instead.",
  },
  Object {
    "level": "error",
    "message": "TriggerTemplate 'template-without-params' references pipeline 'pr-pipeline', but the referenced pipeline cannot be found.",
  },
  Object {
    "level": "error",
    "message": "TriggerTemplate 'template-pipelineref-null' references pipeline 'pipelineref-missing', but the referenced pipeline cannot be found.",
  },
  Object {
    "level": "error",
    "message": "TriggerTemplate 'template-without-pipelineref' references pipeline 'null', but the referenced pipeline cannot be found.",
  },
  Object {
    "level": "error",
    "message": "TriggerTemplate 'template-with-null-pipelinerun-params' references pipeline 'null', but the referenced pipeline cannot be found.",
  },
  Object {
    "level": "error",
    "message": "Invalid name for TriggerTemplate 'template-with-serviceAccount'. Names should be in lowercase, alphanumeric, kebab-case format.",
  },
  Object {
    "level": "warning",
    "message": "Missing base image version 'ibmcom/pipeline-base-image' for step 'fetch-git-token' in Task 'task-with-params-and-inputs'. Specify the base image version, so Tasks can be consistent, and preferably immutable",
  },
  Object {
    "level": "warning",
    "message": "Missing base image version 'ibmcom/pipeline-base-image' for step 'fetch-git-token' in Task 'task-without-params'. Specify the base image version, so Tasks can be consistent, and preferably immutable",
  },
  Object {
    "level": "warning",
    "message": "Missing base image version 'ibmcom/pipeline-base-image' for step 'fetch-git-token' in Task 'task-null-params'. Specify the base image version, so Tasks can be consistent, and preferably immutable",
  },
  Object {
    "level": "warning",
    "message": "Missing base image version 'ibmcom/pipeline-base-image' for step 'fetch-git-token' in Task 'task-without-volumes'. Specify the base image version, so Tasks can be consistent, and preferably immutable",
  },
  Object {
    "level": "warning",
    "message": "Missing base image version 'ibmcom/pipeline-base-image' for step 'fetch-git-token' in Task 'task-with-undefined-volume'. Specify the base image version, so Tasks can be consistent, and preferably immutable",
  },
  Object {
    "level": "warning",
    "message": "Task 'task-with-params-and-inputs' defines parameter 'continuous-delivery-context-secret', but it's not used anywhere in the task spec",
  },
  Object {
    "level": "warning",
    "message": "Task 'task-with-params-and-inputs' defines parameter 'repository-integration', but it's not used anywhere in the task spec",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'ibmcloud-api' at spec.steps[0].env[1].value in 'task-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'repository' at spec.steps[0].env[2].value in 'task-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'propertiesFile' at spec.steps[0].env[3].value in 'task-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'credentials-pvc' at spec.volumes[1].persistentVolumeClaim.claimName in 'task-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'ibmcloud-api-key' at spec.stepTemplate.env[0].value in 'task-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'ibmcloud-api' at spec.steps[0].env[1].value in 'task-without-volumes'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'repository' at spec.steps[0].env[2].value in 'task-without-volumes'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'propertiesFile' at spec.steps[0].env[3].value in 'task-without-volumes'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'ibmcloud-api-key' at spec.stepTemplate.env[0].value in 'task-without-volumes'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'ibmcloud-api' at spec.steps[0].env[1].value in 'task-with-undefined-volume'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'repository' at spec.steps[0].env[2].value in 'task-with-undefined-volume'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'propertiesFile' at spec.steps[0].env[3].value in 'task-with-undefined-volume'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'credentials-pvc' at spec.volumes[0].persistentVolumeClaim.claimName in 'task-with-undefined-volume'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'ibmcloud-api-key' at spec.stepTemplate.env[0].value in 'task-with-undefined-volume'",
  },
  Object {
    "level": "warning",
    "message": "Task 'tekton-alpha-mixed' defines parameter 'my-param', but it's not used anywhere in the task spec",
  },
  Object {
    "level": "warning",
    "message": "Task 'tekton-beta-mixed' defines parameter 'my-param', but it's not used anywhere in the task spec",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'my-missing-param' at spec.steps[0].env[1].value in 'my-alpha-task'",
  },
  Object {
    "level": "warning",
    "message": "Task 'my-alpha-task' defines parameter 'my-unused-param', but it's not used anywhere in the task spec",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'my-missing-param' at spec.steps[0].env[1].value in 'my-beta-task'",
  },
  Object {
    "level": "warning",
    "message": "Task 'my-beta-task' defines parameter 'my-unused-param', but it's not used anywhere in the task spec",
  },
  Object {
    "level": "error",
    "message": "Task 'task-without-volumes' wants to mount volume 'config-volume' in step 'fetch-git-token', but this volume is not defined.",
  },
  Object {
    "level": "error",
    "message": "Task 'task-without-volumes' wants to mount volume 'task-volume' in step 'fetch-git-token', but this volume is not defined.",
  },
  Object {
    "level": "error",
    "message": "Task 'task-with-undefined-volume' wants to mount volume 'config-volume' in step 'fetch-git-token', but this volume is not defined.",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'target-branch' at resourceTemplate.spec.params[0].value in 'pipelinerun-$(uid)'",
  },
  Object {
    "level": "warning",
    "message": "TriggerTemplate 'template-without-resource-templates' defines parameter 'target-branch', but it's not used anywhere in the resourceTemplates specs",
  },
  Object {
    "level": "warning",
    "message": "TriggerTemplate 'template-without-pipelinerun-params' defines parameter 'target-branch', but it's not used anywhere in the resourceTemplates specs",
  },
  Object {
    "level": "warning",
    "message": "TriggerTemplate 'template-with-null-pipelinerun-params' defines parameter 'target-branch', but it's not used anywhere in the resourceTemplates specs",
  },
  Object {
    "level": "warning",
    "message": "TriggerTemplate 'template-with-secret' defines parameter 'target-branch', but it's not used anywhere in the resourceTemplates specs",
  },
  Object {
    "level": "warning",
    "message": "TriggerTemplate 'template-with-serviceAccount' defines parameter 'target-branch', but it's not used anywhere in the resourceTemplates specs",
  },
  Object {
    "level": "error",
    "message": "EventListener 'listener-test' defines trigger template 'template-missing', but the trigger template is missing.",
  },
  Object {
    "level": "error",
    "message": "EventListener 'listener-3' defines trigger template 'template-null-params', but the trigger template is missing.",
  },
  Object {
    "level": "error",
    "message": "EventListener 'listener-5' defines trigger template 'template-with-null-resource-templates', but the trigger template is missing.",
  },
  Object {
    "level": "error",
    "message": "EventListener 'listener-test' defines trigger binding 'binding-missing', but the trigger binding is missing.",
  },
  Object {
    "level": "error",
    "message": "TriggerBinding 'binding-params-duplicate' has param 'revision' duplicated.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-without-params' uses task 'clone' (referenced as 'git-clone'), and it depends on 'get-git-credentials', which doesn't exists (declared in runAfter)",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-task-missing-runafter' uses task 'bar' (referenced as 'bar-task'), and it depends on 'runafter-test', which doesn't exists (declared in runAfter)",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-task-with-null-runafter' uses task 'bar' (referenced as 'bar-task'), and it depends on 'null', which doesn't exists (declared in runAfter)",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-taskspec-invalid-runafters' uses task 'my-task-1' (defined in-line), and it depends on 'this-task-is-not-present', which doesn't exists (declared in runAfter)",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-taskspec-invalid-runafters' uses task 'my-task-1' (defined in-line), and it depends on itself (declared in runAfter)",
  },
  Object {
    "level": "warning",
    "message": "Invalid name for 'statuses_base_url' at spec.steps[0].params[2].value in 'pipeline-with-params'. Names should be in lowercase, alphanumeric, kebab-case format.",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-params' defines parameter 'statuses_url', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-params' defines parameter 'repository', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-params' defines parameter 'branch', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-params' defines parameter 'origin', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-params' defines parameter 'revision', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'pipeline-pvc' at spec.steps[0].params[0].value in 'pipeline-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'ibm-cloud-api-key' at spec.steps[0].params[1].value in 'pipeline-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'statuses_base_url' at spec.steps[0].params[2].value in 'pipeline-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'pipeline-pvc' at spec.steps[1].params[0].value in 'pipeline-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'repository' at spec.steps[1].params[1].value in 'pipeline-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'branch' at spec.steps[1].params[2].value in 'pipeline-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'revision' at spec.steps[1].params[4].value in 'pipeline-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'statuses_url' at spec.steps[2].params[2].value in 'pipeline-without-params'",
  },
  Object {
    "level": "error",
    "message": "Undefined param 'pipeline-pvc' at spec.steps[2].params[3].value in 'pipeline-without-params'",
  },
  Object {
    "level": "warning",
    "message": "Invalid name for 'statuses_base_url' at spec.steps[0].params[2].value in 'pipeline-without-params'. Names should be in lowercase, alphanumeric, kebab-case format.",
  },
  Object {
    "level": "warning",
    "message": "Invalid name for 'fail_task' at spec.steps[2].params[1].name in 'pipeline-without-params'. Names should be in lowercase, alphanumeric, kebab-case format.",
  },
  Object {
    "level": "warning",
    "message": "Invalid name for 'statuses_url' at spec.steps[2].params[2].name in 'pipeline-without-params'. Names should be in lowercase, alphanumeric, kebab-case format.",
  },
  Object {
    "level": "warning",
    "message": "Invalid name for 'statuses_url' at spec.steps[2].params[2].value in 'pipeline-without-params'. Names should be in lowercase, alphanumeric, kebab-case format.",
  },
  Object {
    "level": "warning",
    "message": "Invalid name for 'statuses_base_url' at spec.steps[0].params[2].value in 'pipeline-with-taskref-without-name'. Names should be in lowercase, alphanumeric, kebab-case format.",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-name' defines parameter 'statuses_url', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-name' defines parameter 'repository', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-name' defines parameter 'branch', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-name' defines parameter 'origin', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-name' defines parameter 'revision', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' defines parameter 'statuses_url', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' defines parameter 'statuses_base_url', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' defines parameter 'pipeline-pvc', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' defines parameter 'ibm-cloud-api-key', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' defines parameter 'repository', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' defines parameter 'branch', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' defines parameter 'origin', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-without-params' defines parameter 'revision', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' defines parameter 'statuses_url', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' defines parameter 'statuses_base_url', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' defines parameter 'pipeline-pvc', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' defines parameter 'ibm-cloud-api-key', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' defines parameter 'repository', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' defines parameter 'branch', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' defines parameter 'origin', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-taskref-with-null-params' defines parameter 'revision', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-missing-and-extra-params' defines parameter 'ok-param', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-missing-and-extra-params' defines parameter 'missing-param', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "warning",
    "message": "Pipeline 'pipeline-with-missing-and-extra-params' defines parameter 'default-param', but it's not used anywhere in the pipeline spec",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references task 'task-with-params' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-without-params' references task 'task-with-params' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-without-params' references task 'git-clone' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-without-params' references task 'yaml-lint' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-null-params' references task 'get-git-credentials' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-name' references task 'null' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-task-missing-runafter' references task 'foo-task' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-task-missing-runafter' references task 'bar-task' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-task-with-null-runafter' references task 'foo-task' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-task-with-null-runafter' references task 'bar-task' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-retries' references task 'my-task' but the referenced task cannot be found. To fix this, include all the task definitions to the lint task for this pipeline.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'tekton-alpha-mixed-pipeline' references task 'my-task', but parameter 'my-param' is not supplied (it's a required param in 'my-task')",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'tekton-beta-mixed-pipeline' references task 'my-task', but parameter 'my-param' is not supplied (it's a required param in 'my-task')",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'my-alpha-pipeline' references task 'my-alpha-task' (as 'my-pipeline-task'), and supplies parameter 'my-extra-param' to it, but it's not a valid parameter",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'my-alpha-pipeline' references task 'my-alpha-task' (as 'my-pipeline-task'), but parameter 'my-missing-pipeline-param' is not supplied (it's a required param in 'my-alpha-task')",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'my-alpha-pipeline' references task 'my-other-pipeline-task', and supplies parameter 'my-extra-param' to it, but it's not a valid parameter",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'my-alpha-pipeline' references task 'my-other-pipeline-task', but parameter 'my-missing-pipeline-param' is not supplied (it's a required param in 'my-other-pipeline-task')",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'my-beta-pipeline' references task 'my-beta-task' (as 'my-pipeline-task'), and supplies parameter 'my-extra-param' to it, but it's not a valid parameter",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'my-beta-pipeline' references task 'my-beta-task' (as 'my-pipeline-task'), but parameter 'my-missing-pipeline-param' is not supplied (it's a required param in 'my-beta-task')",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'my-beta-pipeline' references task 'my-other-pipeline-task', and supplies parameter 'my-extra-param' to it, but it's not a valid parameter",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'my-beta-pipeline' references task 'my-other-pipeline-task', but parameter 'my-missing-pipeline-param' is not supplied (it's a required param in 'my-other-pipeline-task')",
  },
  Object {
    "level": "warning",
    "message": "TriggerTemplate 'template-with-params' references pipeline 'pipeline-with-params', and supplies 'target-branch', but it's not a valid parameter.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references param 'statuses_url', but it is not supplied in triggerTemplate 'template-with-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references param 'statuses_base_url', but it is not supplied in triggerTemplate 'template-with-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references param 'pipeline-pvc', but it is not supplied in triggerTemplate 'template-with-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references param 'ibm-cloud-api-key', but it is not supplied in triggerTemplate 'template-with-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references param 'repository', but it is not supplied in triggerTemplate 'template-with-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references param 'branch', but it is not supplied in triggerTemplate 'template-with-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references param 'origin', but it is not supplied in triggerTemplate 'template-with-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-params' references param 'revision', but it is not supplied in triggerTemplate 'template-with-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-params' references param 'statuses_url', but it is not supplied in triggerTemplate 'template-without-pipelinerun-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-params' references param 'statuses_base_url', but it is not supplied in triggerTemplate 'template-without-pipelinerun-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-params' references param 'pipeline-pvc', but it is not supplied in triggerTemplate 'template-without-pipelinerun-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-params' references param 'ibm-cloud-api-key', but it is not supplied in triggerTemplate 'template-without-pipelinerun-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-params' references param 'repository', but it is not supplied in triggerTemplate 'template-without-pipelinerun-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-params' references param 'branch', but it is not supplied in triggerTemplate 'template-without-pipelinerun-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-params' references param 'origin', but it is not supplied in triggerTemplate 'template-without-pipelinerun-params'",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-taskref-without-params' references param 'revision', but it is not supplied in triggerTemplate 'template-without-pipelinerun-params'",
  },
  Object {
    "level": "warning",
    "message": "TriggerTemplate 'template-for-pipeline-params-tests' references pipeline 'pipeline-with-missing-and-extra-params', and supplies 'extra-param', but it's not a valid parameter.",
  },
  Object {
    "level": "error",
    "message": "Pipeline 'pipeline-with-missing-and-extra-params' references param 'missing-param', but it is not supplied in triggerTemplate 'template-for-pipeline-params-tests'",
  },
  Object {
    "level": "error",
    "message": "TriggerTemplate 'my-trigger-template-with-workspace' references Pipeline 'my-pipeline-with-workspace', but provides no workspace for 'missing-pipeline-workspace' (it's a required workspace in 'my-pipeline-with-workspace')",
  },
  Object {
    "level": "warning",
    "message": "TriggerBinding 'binding-params-without-value' defines parameter 'revision' with missing value",
  },
]
`;
